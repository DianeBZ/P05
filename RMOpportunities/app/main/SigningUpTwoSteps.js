import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  Dimensions,
  TextInput,
  TouchableHighlight,
  BackAndroid
} from 'react-native';

import Translation from './Translation'
import SigningUpDone from './SigningUpDone'

var width = Dimensions.get('window').width;
var height = Dimensions.get('window').height;
var star = '*';
// Variables to display only one step at the same time
var step1 = 'false';
var step2 = 'false';

class SigningUpSteps extends Component{
    constructor(props){
        super(props);
        this.state = {
            companyName : '',
            companyAddr: '',
            AddInfoAddr: '',
            city: '',
            zipCode: '',
            country: '',
            VATnumber: '',
            contactName: '',
            mailAddr: '',
            phone: '',
            passwd: '',
        };
    }

    render(){
        return(
        <View>
            {(()=>{
                if(step1==='true'){
                    return(
                        <View>
                        <TouchableHighlight onPress={this._onPressStep1} style={styles.inscription} underlayColor='#FFFFFF'>
                        <View style={{flexDirection:'row'}}>
                            <Text style={{color:"#A4D04A", fontWeight:'bold', fontSize:24}}> 1 </Text>
                            <Text style={{color:'#000000', fontSize:24}}>{Translation[lang].info_entr}</Text>
                        </View>
                        </TouchableHighlight>
                        <View style={{width: width*0.95, alignItems:'center'}}>
                          <View style={{width: width*0.8, alignItems:'flex-start'}}>
                            <TextInput style={styles.textToFill} underlineColorAndroid={'transparent'} placeholder={Translation[lang].nom + star} onChangeText={(companyName) => this.setState({companyName})} value={this.state.companyName}/>
                            <View style={{flexDirection:'row'}}>
                                <TextInput style={[styles.textToFill2, {width:width*0.4}]} underlineColorAndroid={'transparent'} placeholder={Translation[lang].adr + star} onChangeText={(companyAddr) => this.setState({companyAddr})} value={this.state.companyAddr}/>
                                <TextInput style={styles.textToFill2} underlineColorAndroid={'transparent'} placeholder={Translation[lang].compl_adr} onChangeText={(AddInfoAddr) => this.setState({AddInfoAddr})} value={this.state.AddInfoAddr}/>
                            </View>
                            <View style={{flexDirection:'row'}}>
                                <TextInput style={[styles.textToFill2, {width:width*0.4}]} underlineColorAndroid={'transparent'} placeholder={Translation[lang].ville + star} onChangeText={(city) => this.setState({city})} value={this.state.city}/>
                                <TextInput style={styles.textToFill2} underlineColorAndroid={'transparent'} placeholder={Translation[lang].code_post + star} onChangeText={(zipCode) => this.setState({zipCode})} value={this.state.zipCode}/>
                            </View>
                            <View style={{flexDirection:'row'}}>
                                <TextInput style={styles.textToFill2} underlineColorAndroid={'transparent'} placeholder={Translation[lang].pays + star} onChangeText={(country) => this.setState({country})} value={this.state.country}/>
                            </View>
                            <TextInput style={styles.textToFill} underlineColorAndroid={'transparent'} placeholder={Translation[lang].tva} onChangeText={(VATnumber) => this.setState({VATnumber})} value={this.state.VATnumber}/>
                          </View>
                        </View>
                        <TouchableHighlight onPress={this._onPressStep2} style={styles.inscription} underlayColor='#FFFFFF'>
                        <View style={{flexDirection:'row'}}>
                            <Text style={{color:"#A4D04A", fontWeight:'bold', fontSize:24}}> 2 </Text>
                            <Text style={{color:'#000000', fontSize:24}}>{Translation[lang].info_contact}</Text>
                        </View>
                        </TouchableHighlight>
                        </View>
                    );
                }else if (step2==='true'){
                    return(
                        <View>
                        <TouchableHighlight onPress={this._onPressStep1}style={styles.inscription} underlayColor='#FFFFFF'>
                        <View style={{flexDirection:'row'}}>
                            <Text style={{color:"#A4D04A", fontWeight:'bold', fontSize:24}}> 1 </Text>
                            <Text style={{color:'#000000', fontSize:24}}>{Translation[lang].info_entr}</Text>
                        </View>
                        </TouchableHighlight>
                        <TouchableHighlight onPress={this._onPressStep2} style={styles.inscription} underlayColor='#FFFFFF'>
                        <View style={{flexDirection:'row'}}>
                            <Text style={{color:"#A4D04A", fontWeight:'bold', fontSize:24}}> 2 </Text>
                            <Text style={{color:'#000000', fontSize:24}}>{Translation[lang].info_contact}</Text>
                        </View>
                        </TouchableHighlight>
                        <View style={{width: width*0.95, alignItems:'center'}}>
                          <View style={{width: width*0.8, alignItems:'flex-start'}}>
                            <TextInput style={styles.textToFill} underlineColorAndroid={'transparent'} placeholder={Translation[lang].nom_contact + star} onChangeText={(contactName) => this.setState({contactName})} value={this.state.contactName}/>
                            <View style={{flexDirection:'row'}}>
                                <TextInput style={[styles.textToFill2, {width:width*0.4}]} underlineColorAndroid={'transparent'} placeholder={Translation[lang].mail + star} onChangeText={(mailAddr) => this.setState({mailAddr})} value={this.state.mailAddr}/>
                                <TextInput style={styles.textToFill2} underlineColorAndroid={'transparent'} placeholder={Translation[lang].tel + star} onChangeText={(phone) => this.setState({phone})} value={this.state.phone}/>
                            </View>
                            <View style={{flexDirection:'row'}}>
                                <TextInput style={styles.textToFill2} secureTextEntry={true} underlineColorAndroid={'transparent'} placeholder={Translation[lang].mdp + star} onChangeText={(passwd) => this.setState({passwd})} value={this.state.passwd}/>
                            </View>
                          </View>
                        </View>
                        </View>
                    );
                } else {
                    return(
                        <View>
                        <TouchableHighlight onPress={this._onPressStep1} style={styles.inscription} underlayColor='#FFFFFF'>
                        <View style={{flexDirection:'row'}}>
                            <Text style={{color:"#A4D04A", fontWeight:'bold', fontSize:24}}> 1 </Text>
                            <Text style={{color:'#000000', fontSize:24}}>{Translation[lang].info_entr}</Text>
                        </View>
                        </TouchableHighlight>
                        <TouchableHighlight onPress={this._onPressStep2} style={styles.inscription} underlayColor='#FFFFFF'>
                        <View style={{flexDirection:'row'}}>
                            <Text style={{color:"#A4D04A", fontWeight:'bold', fontSize:24}}> 2 </Text>
                            <Text style={{color:'#000000', fontSize:24}}>{Translation[lang].info_contact}</Text>
                        </View>
                        </TouchableHighlight>
                        </View>
                    );
                }
            })()}
        </View>
        );
    }
    
    _onPressStep1 = () =>{
    if(step1 === 'true'){
      step1 = 'false';
    }else{
      step1 = 'true';
      step2 = 'false';
    }
    this.forceUpdate();
  };
  
  _onPressStep2 = () =>{
    if(step2 === 'true'){
      step2 = 'false';
    }else{
      step2 = 'true';
      step1 = 'false';
    }
    this.forceUpdate();
  };
}

export default class SigningUpTwoSteps extends Component {
    render(){
        BackAndroid.addEventListener('Back',this.onBackAndroid);
        return(
        <View style={{flex:1}}>
            <View style={{height: height * 0.08}}/>
            <View style={styles.window}>
                <Text style={styles.titre}>
                    {Translation[lang].inscript}
                </Text>
                <View style={styles.mainWindow}>
                    <SigningUpSteps/>
                    <TouchableHighlight onPress={this.onPressInscription} style={[styles.button, {width: width *0.5, backgroundColor:'#A4D04A', marginBottom:20}]} underlayColor="rgb(138,183,46)">
                        <Text style={styles.buttonText}>{Translation[lang].sinscrire}</Text>
                    </TouchableHighlight>
                    <View style={{alignItems:'center', width:width*0.9}}>
                        <Text numberOfLines={4}>
                            <Text style={{color:'#000000'}}>{Translation[lang].texte_cgu1}</Text>
                            <Text style={{color:'#A4D04A'}}>{Translation[lang].texte_cgu2}</Text>
                            <Text style={{color:'#000000'}}>{Translation[lang].texte_cgu3}</Text>
                        </Text>
                    </View>
                </View>
            </View>
        </View>
        );
    }

    onPressInscription = () => {
		BackAndroid.removeEventListener('Exit',this.onBackAndroid);
		const { navigator } = this.props;
		if (navigator) {
			navigator.push({
				name: 'SigningUpDone',
				component: SigningUpDone,
			})
		}
	};
    
    onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
    window:{
        width: width,
        height: height*0.93,
        backgroundColor: '#F2F2F2',
        alignItems: 'center',
    },
    mainWindow:{
        width: width,
        height: height,
        backgroundColor: "#FFFFFF",
        alignItems: 'center',
    },
    titre:{
        fontSize: 32,
        textAlign: 'center',
        color: '#000000',
        marginTop: 15,
        marginBottom: 10
    },
    button:{
        height: height*0.05,
        marginTop: 20,
        borderRadius:5,
        justifyContent: 'center',
    },
    buttonText:{
        color: "white",
        textAlign: "center",
        fontSize: 20,
    },
    inscription:{
        backgroundColor:'#FFFFFF',
        width:width*0.95,
        alignItems:'flex-start',
        marginTop:20
    },
    textToFill:{
       width: width*0.8,
       height: height* 0.06,
       borderWidth: 1,
       borderColor: "#000000",
       marginTop: 15,
       color: 'grey',
       borderRadius: 3,
   },
   textToFill2:{
       width: width*0.35,
       height: height* 0.06,
       borderWidth: 1,
       borderColor: "#000000",
       marginTop: 15,
       marginRight: width*0.05,
       color: 'grey',
       borderRadius: 3,
   },
});
