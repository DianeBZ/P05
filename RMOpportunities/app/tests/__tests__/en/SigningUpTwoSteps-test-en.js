// __tests__/SigningUpThreeSteps-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import SigningUpTwoSteps from '../../../main/SigningUpTwoSteps';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <SigningUpTwoSteps />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
