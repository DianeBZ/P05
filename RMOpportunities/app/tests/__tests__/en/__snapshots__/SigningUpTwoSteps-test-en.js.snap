// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 106.72,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F2F2F2",
        "height": 1240.6200000000001,
        "width": 750,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontSize": 32,
          "marginBottom": 10,
          "marginTop": 15,
          "textAlign": "center",
        }
      }
    >
      Sign up
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFFFF",
          "height": 1334,
          "width": 750,
        }
      }
    >
      <View>
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFFFFF",
                  "marginTop": 20,
                  "width": 712.5,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#A4D04A",
                    "fontSize": 24,
                    "fontWeight": "bold",
                  }
                }
              >
                 1 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  }
                }
              >
                Company information
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "alignItems": "flex-start",
                  "backgroundColor": "#FFFFFF",
                  "marginTop": 20,
                  "width": 712.5,
                },
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#A4D04A",
                    "fontSize": 24,
                    "fontWeight": "bold",
                  }
                }
              >
                 2 
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000000",
                    "fontSize": 24,
                  }
                }
              >
                Representative information
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "borderRadius": 5,
                "height": 66.7,
                "justifyContent": "center",
                "marginTop": 20,
              },
              Object {
                "backgroundColor": "#A4D04A",
                "marginBottom": 20,
                "width": 375,
              },
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          Get started
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "width": 675,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={4}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
              }
            }
          >
            By continuing you acknowledge that you have read the 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#A4D04A",
              }
            }
          >
             General terms and conditions of RMOpportunities
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000000",
              }
            }
          >
            , that the informations entered are correct and that you will update them if necessary.
          </Text>
        </Text>
      </View>
    </View>
  </View>
</View>
`;
